packages/flutter/lib/src/widgets/form.dart

```diff
@@ -163,7 +163,8 @@ class FormState extends State<Form> {
 
   @override
   Widget build(BuildContext context) {
-    if (widget.autovalidate)
+    // validate only if one of the text form fields changes
+    if (widget.autovalidate && _generation > 0)
```

Comment:
I think this is more complex than i thought, the reset method will not reset the _generation but increment it by 1. The documentation of that method actually mentioned the field will gets revalidate if autovalidate is true, but that is contradict with the formfield.reset.

Reply:
This make sense. 

I think form.reset should not validate de formFields, this may cause what we are trying to fix: validate the without user iteraction